pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,169 @@
+        shallowCopy.remove("serviceSecretKey");
+        shallowCopy.put("service-key", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretValue");
+        shallowCopy.put("service-val", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("brewBuilds");
+        shallowCopy.put("brew_builds", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmUrl");
+        shallowCopy.put("SCMURL", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmRevision");
+        shallowCopy.put("scm-revision", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraScmUrls");
+        shallowCopy.put("xtraScmUrls", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraBuilds");
+        shallowCopy.put("extraBuild_", null);
+        negativeTest(shallowCopy);
+    }
+
+    void negativeTest(HashMap<String, Object> testMap) {
+        try {
+            scanService.validateMapKeys(testMap);
+            assert (false);
+        } catch (Exception e) {
+            assert (true);
+        }
+    }
+
+    @Test
+    void positiveMandatoryKeywordsTest() {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put("productID", 111);
+        map.put("serviceUrl", "http://someurl.com:8080");
+        map.put("serviceSecretKey", "pig");
+        map.put("serviceSecretValue", "sometoken");
+        try {
+            scanService.checkPresenceOfMandatoryFields(map);
+            assert (true);
+        } catch (Validate.ConfigMissingException e) {
+            assert (false);
+        }
+    }
+
+    @Test
+    void negativeMandatoryKeywordsTest() {
+
+        for (String mk : minimalConfigMap.keySet()) {
+            HashMap<String, Object> map = new HashMap<>();
+            map.remove(mk);
+            try {
+                scanService.checkPresenceOfMandatoryFields(map);
+                assert (false);
+            } catch (Validate.ConfigMissingException e) {
+                assert (true);
+            }
+        }
+    }
+
+    void showMap(Map<String, ?> map) {
+        try {
+            for (Map.Entry<String, ?> entry : map.entrySet()) {
+                System.out.printf("%30s : %s%n", entry.getKey(), entry.getValue());
+            }
+            System.out.println(new String(new char[5]).replace("\0", "========="));
+        } catch (Exception e) {
+            log.error(e.getMessage());
```

Comment:
Mixture of logging and system.out/err.

Reply:
log.error has been replaced by System.out.println